### PR TITLE
Make auto-instr work with latest otel-java snapshot

### LINKED ISSUE #363: Cannot build against otel-java 0.4.0-snapshot
It seems that Otel-java has several changes since last release of 0.3.0 that broke auto-instr. It currently does not compile against latest 0.4.0 snapshot. E.g. [NoopScope was removed](https://github.com/open-telemetry/opentelemetry-java/pull/1062) and several others.

This also raises the question about regular (nightly?) cross-compilation between the two repos. How we detect such breaking changes and fix their consequences?

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #363

### CODE REVIEW

File: gradle/dependencies.gradle [hunk 1]
```diff
@@ -1,9 +1,13 @@
 def groovyVer = "2.5.11"
 def spockGroovyVer = groovyVer.replaceAll(/\.\d+$/, '')
 
+configurations.all {
+  resolutionStrategy.cacheChangingModulesFor 12, 'hours'
+}
```
Comment (trask) [inline gradle/dependencies.gradle:6]: so we can search for these comments when merging to master
```suggestion
// this is only needed for the depends-on-otel-java-snapshot branch
configurations.all {
  resolutionStrategy.cacheChangingModulesFor 12, 'hours'
}
```

File: instrumentation/hystrix-1.4/hystrix-1.4.gradle [hunk 2]
```diff
@@ -26,6 +26,6 @@ dependencies {
   testCompile group: 'io.reactivex', name: 'rxjava', version: '1.0.7'
   testCompile group: 'com.netflix.hystrix', name: 'hystrix-core', version: '1.4.0'
 
-  latestDepTestCompile group: 'io.reactivex', name: 'rxjava', version: '+'
+  latestDepTestCompile group: 'io.reactivex', name: 'rxjava', version: '1.+'
```
[the remark this answers is not in the document]
Comment (iNikem) [reply instrumentation/hystrix-1.4/hystrix-1.4.gradle:29]: This is not exactly true. Adding oss repo brought rxjava 2.0-SNAPSHOT which broke `latestDepTest`. But that just shows that we really don't support (unreleased) 2.0 yet, so we it is Ok to explicitly limit dependency version to `1.+`

[the remark this answers is not in the document]
Comment (trask) [reply instrumentation/hystrix-1.4/hystrix-1.4.gradle:29]: If we bring this change into master, then our build won't fail if/when 2.0 is released, which helps us to know that we need to go and update our instrumentation to work with the new version (or to document that we do not support the new version yet).

[the remark this answers is not in the document]
Comment (iNikem) [reply instrumentation/hystrix-1.4/hystrix-1.4.gradle:29]: Good point. And I found a better solution :) I limit oss repo only to `io.opentelemetry` artifacts.

**Comment by iNikem** [comment]
@trask can you please rerun CI build for this?

**Comment by trask** [comment]
Looks like `muzzle` is failing. Since this isn't going directly to master, I'm ok merging if you want. Just let me know.

**Comment by iNikem** [comment]
@trask I will take care of `muzzle`. But `otel-java` 0.4.0 was released yesterday. Should we now switch to released version and aim to merge this PR into `master`?

**Comment by trask** [comment]
Yes, that sounds great

**Comment by iNikem** [comment]
@trask changed target branch. `muzzle` is Ok now, but another job failed :(

**Pull request merged**